onnx/common/ir_pb_converter.cc

```diff
@@ -533,9 +538,62 @@ void encodeGraph(ONNX_NAMESPACE::GraphProto * p_g, const std::shared_ptr<Graph>
   }
 }
 
-void ExportModelProto(ONNX_NAMESPACE::ModelProto* p_m, const std::shared_ptr<Graph>& g) {
-  ONNX_NAMESPACE::GraphProto* p_g = p_m->mutable_graph();
+void ExportModelProto(ModelProto* p_m, const std::shared_ptr<Graph>& g) {
+  GraphProto* p_g = p_m->mutable_graph();
   encodeGraph(p_g, g);
+  // Add new opset_versions
+  p_m->clear_opset_import();
+  for (const OpSetID& opset : g->opset_versions_mutable()) {
+    OperatorSetIdProto *opset_version_output = p_m->add_opset_import();
+    opset_version_output->set_domain(opset.domain());
+    opset_version_output->set_version(opset.version());
+  }
+}
+
+// TODO (Optimizer): Should this just be copied instead?
```

Comment:
Optimization: probably do it in place

Reply:
Realized that this introduces challenges with over-writing existing nodes/inputs/outputs, so creating the new ModelProto makes a lot of sense, rather than messing with large existing ModelProtos.